Add JobPreview / Standardize default page margins / Convert NavBar to grid / General CSS standardisation adjustments

zhenyangg opened the pull request:
1. Added JobPreview:
<img width="1440" alt="Screenshot 2022-01-04 at 12 22 45 PM" src="https://user-images.githubusercontent.com/15624933/148009400-5c5c6683-da32-441e-8cdc-3ba657d6d860.png">

2. Reduced padding on JobCard to match wireframe more accurately
3. Decomposed content within links in JobCard for reusability in JobPreview and future components

carlomigueldy commented:
Hey awesome stuff man @zhenyangg thanks for contributing! 🤝  Would love to get some context like which Figma designs you were basing this off, or which issue was it? You can mention the issue from the description btw.

carlomigueldy commented:
@allcontributors please add @zhenyangg for code!

zhenyangg commented:
> Hey awesome stuff man @zhenyangg thanks for contributing! 🤝 Would love to get some context like which Figma designs you were basing this off, or which issue was it? You can mention the issue from the description btw.

Here's the issue: https://github.com/Developer-DAO/DAO-job-board/issues/190
Worked on: JobPreview component (right side of the screen)
and the screenshot from Figma: 
![IN PROGRESS - Jobs board -  Create New Job v1](https://user-images.githubusercontent.com/15624933/148322880-48245ef8-9869-4990-8794-d77a1c09318a.png)

zhenyangg commented:
More updates:
1. Replaced sidebar fontawesome icons with tabler icons
2. Adjusted sidebar paddings to reflect new design changes
3. Added 'Create new job' button and discord link

Update preview (shadow not included
<img width="287" alt="Screenshot 2022-01-06 at 6 20 57 PM" src="https://user-images.githubusercontent.com/15624933/148368021-167c3cb3-40bf-45ac-9d93-cdac7522fac7.png">

PBillingsby commented:
@carlomigueldy how do we get this deployed so I can test the PR app?

carlomigueldy commented:
> @carlomigueldy how do we get this deployed so I can test the PR app?

Oh, about that we can't authenticate it via Vercel since we aren't part of Vercel Team.

Unsure if @angeljgomezc is already part of Developer DAO Vercel team though.

angeljgomezc commented:
> > @carlomigueldy how do we get this deployed so I can test the PR app?
> 
> Oh, about that we can't authenticate it via Vercel since we aren't part of Vercel Team.
> 
> Unsure if @angeljgomezc is already part of Developer DAO Vercel team though.

I haven't checked this yet because I'm looking for a way to gain access to the Vercel or get a Vercel sponsorship before moving forward

It's cool to review code and stuff, but having our PRs deployed would be much better - so I'm waiting for that before merging any front-end

angeljgomezc commented:
Hey @zhenyangg ! Thanks for this work - it looks great!

I'd appreciate if you could close this PR and reopen a new one so we can deploy this before merging :)

PBillingsby commented:
@zhenyangg Need these conflicts resolved and I will test

zhenyangg commented:
> @zhenyangg Need these conflicts resolved and I will test

conflicts resolved.

angeljgomezc reviewed (changes requested):
Hey mate! I have a few things I'd like you to fix before we merge this again:

1. We recently pushed some code with Storybook components/design patterns that we'd like to follow. If you perform yarn storybook on the main folder, you'll get the page that shows everything we're trying to use from here, from buttons to colors, text, headings, etc. Ask us any questions about how to use these variants (should probably add this to the contribution guidelines).

2. Even though we weren't trying to make the site responsive at first, I think it's still important to make it as responsive as possible. So, it'd be great if you could fix some of the code to be a little more responsive and match better with the designs on Figma.

Once you get these done, we'll review again and merge if necessary!

Sorry if we're asking too much of you, but we're trying to get this working as polished as possible so we don't have to refactor too much later.

Thanks for your help anyway! This is a great contribution - we would just like to make it a bit better :)

zhenyangg commented:
1. I'm using the existing components.
2. Will review and update. Noticed lots of discrepancies from the wireframe on figma, thus i've been editing the other components unrelated to JobPreview - the global ones such as JobCard, NavBar. Appreciate if you can push this PR asap as I have to keep merging the main branch's code with the global components I updated on my side.

zhenyangg commented:
Mobile breakpoints added for JobPreview, NavBar, and PostJob page (post-job > index.tsx
<img width="287" alt="Screenshot 2022-01-06 at 6 20 57 PM" src="https://user-images.githubusercontent.com/15624933/148896889-7f614237-c31a-4936-ade0-65efb3c5b762.png">
<img width="366" alt="Screenshot 2022-01-11 at 2 44 47 PM" src="https://user-images.githubusercontent.com/15624933/148896899-7f16e229-bd14-41b4-b292-e68878661b77.png">
<img width="352" alt="Screenshot 2022-01-11 at 3 05 29 PM" src="https://user-images.githubusercontent.com/15624933/148896905-04378f64-cc02-4e3b-b38f-328229983b44.png">
)

zhenyangg commented:
1. Standardized margins across all existing pages
2. Fix hover behaviour; changed margin to transform on DeveloperItem & JobCard hovers
3. Reduced DeveloperItem & JobCard hover on y-axis
3. Hide "+ Create new job" button except for hirers

angeljgomezc commented on packages/client/components/navigation/Sidebar.tsx:
```diff
@@ -152,24 +146,66 @@ export default function Dashboard() {
+                <Button
+                  color="white"
+                  colorScheme="neutral"
+                  variant="morePadding"
```
Hey @zhenyangg ! We're trying to standardize the use of the button variants across the site instead of adding so many props. Can you use `neutral` variant instead? But great job anyways, this actually looks cool

zhenyangg replied:
As discussed, let's leave this as it is for now, since there's no way to combine multiple variants.

angeljgomezc commented:
> This pull request is being automatically deployed with Vercel ([learn more](https://vercel.link/github-learn-more)). To see the status of your deployment, click below or on the icon next to each commit.
> 
> 🔍 Inspect: https://vercel.com/devdaojobboard/dao-job-board/6btjDL5XUHbhTjScKfKVLn7zegzx ✅ Preview: https://dao-job-board-git-fork-zhenyangg-main-devdaojobboard.vercel.app

Hey @zhenyangg ! The changes look awesome - just checked the preview and there are still a few things to polish:

1. The input for changing user mode has a border/outline on focus. Can you get rid of that? It looks awkward. 
![image](https://user-images.githubusercontent.com/53134194/149014641-1a97c741-fe40-48aa-996d-5a24d9bf6200.png)

2. The `JobPost` modal in the `JobList` page doesn't open, it goes directly into the job page. We're trying to make it different, like this: https://devdao-job-board.vercel.app/jobs (click on a job so you can see). You may need to rebase/pull from the repo in that case.

Apart from that, all changes look good. We're all happy with your contributions - thank you!

zhenyangg commented:
> 3\. so

1. Reset to original.
2. Removed placeholder.
3. Fixed expected behaviour of modal.

zhenyangg commented:
> > This pull request is being automatically deployed with Vercel ([learn more](https://vercel.link/github-learn-more)). To see the status of your deployment, click below or on the icon next to each commit.
> > 🔍 Inspect: https://vercel.com/devdaojobboard/dao-job-board/6btjDL5XUHbhTjScKfKVLn7zegzx ✅ Preview: https://dao-job-board-git-fork-zhenyangg-main-devdaojobboard.vercel.app
> 
> Hey @zhenyangg ! The changes look awesome - just checked the preview and there are still a few things to polish:
> 
> 1. The input for changing user mode has a border/outline on focus. Can you get rid of that? It looks awkward.
>    ![image](https://user-images.githubusercontent.com/53134194/149014641-1a97c741-fe40-48aa-996d-5a24d9bf6200.png)
> 2. Also, the select shouldn't show a placeholder. You can check the current deployment from https://devdao-job-board.vercel.app/ to have an idea. I'm afraid you may need to rebase/pull from the repo in that case, so these changes don't get pushed into the main branch.
> 3. The `JobPost` modal in the `JobList` page doesn't open, it goes directly into the job page. We're trying to make it different, like this: https://devdao-job-board.vercel.app/jobs (click on a job so you can see).
> 
> Apart from that, all changes look good. We're all happy with your contributions - thank you!

1. Reset to original.
2. Removed placeholder.
3. Fixed expected behaviour of modal.

angeljgomezc commented:
> > > This pull request is being automatically deployed with Vercel ([learn more](https://vercel.link/github-learn-more)). To see the status of your deployment, click below or on the icon next to each commit.
> > > 🔍 Inspect: https://vercel.com/devdaojobboard/dao-job-board/6btjDL5XUHbhTjScKfKVLn7zegzx ✅ Preview: https://dao-job-board-git-fork-zhenyangg-main-devdaojobboard.vercel.app
> > 
> > 
> > Hey @zhenyangg ! The changes look awesome - just checked the preview and there are still a few things to polish:
> > 
> > 1. The input for changing user mode has a border/outline on focus. Can you get rid of that? It looks awkward.
> >    ![image](https://user-images.githubusercontent.com/53134194/149014641-1a97c741-fe40-48aa-996d-5a24d9bf6200.png)
> > 2. Also, the select shouldn't show a placeholder. You can check the current deployment from https://devdao-job-board.vercel.app/ to have an idea. I'm afraid you may need to rebase/pull from the repo in that case, so these changes don't get pushed into the main branch.
> > 3. The `JobPost` modal in the `JobList` page doesn't open, it goes directly into the job page. We're trying to make it different, like this: https://devdao-job-board.vercel.app/jobs (click on a job so you can see).
> > 
> > Apart from that, all changes look good. We're all happy with your contributions - thank you!
> 
>     1. Reset to original.
> 
>     2. Removed placeholder.
> 
>     3. Fixed expected behaviour of modal.

AWESOME WORK! Thank you!

zhenyangg commented:
yay first commit to main :)
